apps/site/lib/site_web/controllers/schedule/line_api.ex

```diff
@@ -87,7 +82,25 @@ defmodule SiteWeb.ScheduleController.LineApi do
       end)
       |> Enum.into(%{})
 
-    json(conn, combined_data_by_stop)
+    Jason.encode!(combined_data_by_stop)
+  end
+
+  @doc """
+  Provides predictions and vehicle information for a given route and direction, organized by stop.
+  The line diagram polls this endpoint for its real-time data.
+  """
+  @spec realtime(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def realtime(conn, %{"id" => route_id, "direction_id" => direction_id}) do
+    realtime_key = [route_id, direction_id, conn.assigns.date]
+
+    payload =
+      ConCache.get_or_store(:line_diagram_realtime_cache, realtime_key, fn ->
+        do_realtime(route_id, direction_id, conn.assigns.date, conn.assigns.date_time)
+      end)
+
+    conn
+    |> put_resp_content_type("application/json")
+    |> send_resp(200, payload)
```

Comment:
Curious why this was changed from `json(conn, payload)`, as it seems to do the same thing

Reply:
`do_realtime` returns a string, as the work it does includes the JSON encoding of the payload. `json(conn, payload)` automatically encodes `payload`, so if we kept using `json`, we'd end up with a doubly-encoded payload.

Reply:
I see. Could `do_realtime` _not_ encode the data? Or is that required for the caching?

Reply:
It's not required, but my reasoning is that JSON-encoding isn't free either, so if we're caching work let's cache that too.

Reply:
Sure. My personal inclination is to "let it encode" given the likely-small time contribution of the encoding process (Jason was developed specifically for speed), but I don't think the discussion is worth holding things up.